src/unit_tests/syscheckd/test_run_check.c

```diff
@@ -219,6 +225,42 @@ static int teardown_max_fps(void **state) {
     return 0;
 }
 
+#ifdef TEST_WINAGENT
+
+typedef struct _win32rtfim {
+    HANDLE h;
+    OVERLAPPED overlap;
+
+    char *dir;
+    TCHAR buffer[65536];
+    unsigned int watch_status;
+} win32rtfim;
+
+extern void free_win32rtfim_data(win32rtfim *data);
+
+static int setup_hash(void **state) {
+    directory_t *dir_it;
+    win32rtfim *rtlocald;
+    rtlocald = calloc(1, sizeof(win32rtfim));
+    foreach_array(dir_it, syscheck.directories) {
+        if (dir_it->options & REALTIME_ACTIVE) {
+            OSHash_Add_ex(syscheck.realtime->dirtb, strdup(dir_it->path), rtlocald);
```

Comment:
The key is hashed and therefore is not needed to be duplicated (this will actually be a leak in the test).
```suggestion
            OSHash_Add_ex(syscheck.realtime->dirtb, dir_it->path, rtlocald);
```